src/Mvc/Mvc.RazorPages/src/Infrastructure/DynamicPageEndpointMatcherPolicy.cs

```diff
@@ -132,7 +134,8 @@ public async Task ApplyAsync(HttpContext httpContext, CandidateSet candidates)
                     continue;
                 }
 
-                var endpoints = _selector.SelectEndpoints(dynamicValues);
+                selector = ResolveSelector(selector, endpoint);
+                var endpoints = selector.SelectEndpoints(dynamicValues);
```

Comment:
Couldn't `selector` can be null here?

Reply:
No. The cache will throw if it can't find a registered instance.